exist-core/src/main/java/org/exist/xquery/PerformanceStats.java

```diff
@@ -19,359 +19,100 @@
+    String XML_ELEMENT_CALLS = "calls";
 
-        public boolean equals(Object obj) {
-        	if (obj != null && obj instanceof IndexStats) {
-                final IndexStats other = (IndexStats) obj;
-                return other.indexType.equals(indexType) && other.source.equals(source) &&
-                    other.line == line && other.column == column && other.mode == mode;
-			}
-        	return false;
-        }
-    }
-
-    private static class QueryStats {
-
-        String source;
-        long executionTime = 0;
-        int callCount = 1;
-
-        QueryStats(String source) {
-            this.source = source;
-            if (this.source == null)
-                {this.source = "";}
-        }
-
-        public void recordCall(long elapsed) {
-            executionTime += elapsed;
-            callCount++;
-        }
-
-        public int hashCode() {
-            return source.hashCode();
-        }
-
-        public boolean equals(Object obj) {
-        	if (obj != null && obj instanceof QueryStats) {
-                return ((QueryStats)obj).source.equals(source);
-			}
-        	return false;
-        }
-    }
+    String RANGE_IDX_TYPE = "range";
 
-    private static class FunctionStats extends QueryStats {
-
-        QName qname;
-
-        FunctionStats(String source, QName name) {
-            super(source);
-            this.qname = name;
-        }
-
-        public int hashCode() {
-            return 31 * qname.hashCode() + source.hashCode();
-        }
-
-        public boolean equals(Object obj) {
-        	if (obj != null && obj instanceof FunctionStats) {
-                final FunctionStats ostats = (FunctionStats) obj;
-                return qname.equals(ostats.qname) &&
-                        source.equals(ostats.source);
-			}
-        	return false;
-        }
-    }
-
-    private static class OptimizationStats {
-
-        String source;
-        OptimizationType type;
-        int line;
-        int column;
-
-        OptimizationStats(String source, OptimizationType type, int line, int column) {
-            if (source == null) {
-                this.source = "";
-            } else {
-                this.source = source;
-            }
-            this.type = type;
-            this.line = line;
-            this.column = column;
-        }
-
-        @Override
-        public int hashCode() {
-            return 32 * type.hashCode() + source.hashCode() + line + column;
-        }
-
-        @Override
-        public boolean equals(Object obj) {
-            if (obj instanceof OptimizationStats) {
-                final OptimizationStats stats = (OptimizationStats) obj;
-                return source.equals(stats.source) && type == stats.type &&
-                        line == stats.line && column == stats.column;
-            }
-            return false;
-        }
-    }
-
-    public enum OptimizationType {
+    enum OptimizationType {
         PositionalPredicate
```

Comment:
This enumeration value should be UPPER in accordance to the java standards.

Reply:
Agreed! Not sure what I was thinking there!

Reply:
Done.